src/ai/backend/common/clients/valkey_client/valkey_schedule/client.py

```diff
@@ -670,6 +684,18 @@ async def check_kernel_presence_status_batch(
         current_time_str = str(current_time)
         batch = Batch(is_atomic=False)
 
+        # Update agent last_check timestamps first
+        agent_ops_count = 0
+        if agent_ids:
+            for agent_id in agent_ids:
+                agent_key = self._get_agent_last_check_key(agent_id)
+                batch.set(
+                    agent_key,
+                    current_time_str,
+                    expiry=ExpirySet(ExpiryType.SEC, AGENT_LAST_CHECK_TTL_SEC),
+                )
+                agent_ops_count += 1
+
```

Comment:
Can we separate this part into other private function?

Reply:
There's only one operation, so I don't think it's necessary.